Shopify/tophat
# Improve handling of Bitrise token editing

### What does this change accomplish?

This change addresses two issues:
1. Editing a Bitrise token would occasionally not propagate the update to the artifact provider.
2. A failed download would be cached as "failed," so fixing the token within the 30 second rolling window would not work unless Tophat was left idle for at least 30 seconds.

### How have you achieved it?

- By making the `@SecureStorage` property wrapper _not_ dynamic.  This prevents it from being bound to the UI, but because writing to the keychain is synchronous, we wouldn't want to wait on updates here anyway.  Synchronous reads are, however, acceptable (especially in artifact providers).
- By fixing an issue in `CachingApplicationDownloader` that caused failed downloads to continue to be returned.  Now, only successful downloads are returned from the cache.  This means that Bitrise downloads can be retried immediately once the token is updated.

### How can the change be tested?

1. Edit your Bitrise token in Tophat to be invalid (e.g. add a 1 to the end).
2. Trigger an app install.
3. Immediately fix the token and retry the download.
4. The install should succeed.
